### ISSUE TITLE
extract_nwis_df wrong variable name

### ISSUE THREAD

**Issue by lapidesd** [mroberge/hydrofunctions#89]
* HydroFunctions version:
* Python version:
* Operating System:

### Description

I ran a hydrofunctions.NWIS query to get all temperature data in the state of California. When converting to a dataframe, there was a division by zero error since the data frequency was miscalculated.

### What I Did to resolve this error

```
I adjusted the source code so that in the extract_nwis_df at lines 571 and 572 of hydrofunctions.py (version on github), I replaced the variable 'freqs' with 'freqs2', which removes the zeros and prevents the division by zero error.
```

**Comment by lapidesd** [comment]
Hi,

I did the query for water temperature ‘00010’. The full query is:

state_data = hf.NWIS(site=None, stateCd=‘CA', service='dv', start_date='1950-01-01', end_date='2021-04-01',
                     parameterCd = '00010')
state_data = state_data.df()

The error shows up on conversion to a dataframe. I just made a pull request for the bug fix. 

Best,
Dana

———————————————————
Dana Lapides, PhD
Wisconsin Water Resources Science-Policy Fellow
UW Madison Aquatic Science Center
Wisconsin Department of Natural Resources
www.danalapides.com <http://www.danalapides.com/>

**Comment by mroberge** [comment]
Excellent!
This query illustrates a problem with how hydrofunctions converts the JSON from the NWIS into a dataframe. It basically takes the data from every single temperature-measuring station in the state of California and resamples it to the frequency of the station with the shortest frequency for the entire length of the query. So if there is a station that recorded temperature every five minutes one day back in 2005, hydrofunctions will output a dataframe with a row for every five minutes starting in 1950 until 2021. And it performs this "service" for you just because you asked for some data.

My plan for the future is to change this so that the data is stored internally in the WaterML format, and a dataframe is created only when you ask for a dataframe. That way the 'upsampling' only occurs if you ask it to occur.

**Comment by mroberge** [comment]
In the meantime, you can see what stations have collected temperature by just requesting the most recent value:
```
hf.NWIS(stateCd="CA", service="dv", parameterCd="00010")
```
You can also use `hf.data_catalog(station_id)` or `hf.site_file(station_id` to learn more about individual stations.

**Comment by mroberge** [comment]
closed with  16dd705
